src/BuildScriptGenerator/Golang/GolangPlatform.cs

```diff
@@ -0,0 +1,278 @@
+            });
+
+            if (detectionResult == null)
+            {
+                return null;
+            }
+
+            ResolveVersions(context, detectionResult);
+            return detectionResult;
+        }
+
+        /// <inheritdoc/>
+        public BuildScriptSnippet GenerateBashBuildScriptSnippet(
+            BuildScriptGeneratorContext ctx,
+            PlatformDetectorResult detectorResult)
+        {
+            // confirm go detector not null
+            var goPlatformDetectorResult = detectorResult as GolangPlatformDetectorResult;
+            if (goPlatformDetectorResult == null)
+            {
+                throw new ArgumentException(
+                    $"Expected '{nameof(detectorResult)}' argument to be of type " +
+                    $"'{typeof(GolangPlatformDetectorResult)}' but got '{detectorResult.GetType()}'.");
+            }
+
+            if (!goPlatformDetectorResult.GoModExists)
+            {
+                throw new InvalidUsageException("No go.mod found at the root of the repo. Please provide a go.mod file containing the version.");
+            }
+
+            // build properties & script snippets
+            var buildProperties = new Dictionary<string, string>();
+
+            // Write platform name and version to the manifest file
+            buildProperties[ManifestFilePropertyKeys.GolangVersion] = goPlatformDetectorResult.PlatformVersion;
+            _logger.LogDebug($"Selected Go version: {goPlatformDetectorResult.PlatformVersion}");
+
+            var scriptProps = new GolangBashBuildSnippetProperties
+            {
+                GoModExists = goPlatformDetectorResult.GoModExists
+            };
+
+            string script = TemplateHelper.Render(
+                TemplateHelper.TemplateResource.GolangSnippet,
+                scriptProps,
+                _logger);
+
+            return new BuildScriptSnippet
+            {
+                BashBuildScriptSnippet = script,
+                BuildProperties = buildProperties
+            };
+        }
+
+        public string GenerateBashRunTimeInstallationScript(RunTimeInstallationScriptGeneratorOptions options)
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool IsEnabled(RepositoryContext ctx)
+        {
+            return _commonOptions.EnableGolangBuild;
+        }
+
+        public bool IsCleanRepo(ISourceRepo repo)
+        {
+            return true;
+        }
+
+        public IEnumerable<string> GetDirectoriesToExcludeFromCopyToBuildOutputDir(BuildScriptGeneratorContext scriptGeneratorContext)
+        {
+            return Array.Empty<string>();
+        }
+
+        public IEnumerable<string> GetDirectoriesToExcludeFromCopyToIntermediateDir(BuildScriptGeneratorContext scriptGeneratorContext)
+        {
+            return Array.Empty<string>();
+        }
+
+        public bool IsEnabledForMultiPlatformBuild(RepositoryContext ctx)
+        {
+            return true;
```

Comment:
I would make it false for sake of simplicity for now

Reply:
Updated